src/devices/Dhtxx/DhtBase.cs

```diff
@@ -280,6 +280,7 @@ public void Dispose()
             if (_shouldDispose)
             {
                 _controller?.Dispose();
+                _controller = null;
```

Comment:
Sorry to insist: this needs to be outside the if (or in both cases). Otherwise, if `shouldDispose` is false, this still bombs on the second `Dispose` call.

Reply:
If ```_shouldDispose``` is false, the key principal is not to dispose. As an example, with the FT4222, you can only create 1 GPIO controller. If you have 2 devices which are using GPIO, and the first one dispose it, then the second one will raise an exception.
So it's important not to dispose at all the GPIO controller if the ```shouldDispose```` flag is false.

Reply:
True. I'm **not** talking about disposing, I'm talking about setting the controller to null. This doesn't do an implicit dispose or something.

Reply:
@pgrawehr I think I really miss something then. Can you please then write the full dispose class how you think it should be for me to understand what you mean.
Again, here is the principle:
- Whatever happens the I2C device is disposed
- the GPIO Controller is disposed only if _shouldDispose is true
- if _shouldDispose is flase, the open pins are closed and the controller is not disposed at all
To prevent some mistakes:
- _shouldDipose is set to true if the GPIO Controller is created in the class. So even if you pass false, if the controller = null in the class constructor, a controller will be created and _shouldDiposed will be set to true

Reply:
I'll do that. It really looks like we're talking different things. But give me some time,  as you probably know,  I'm traveling around,  and good WiFi is hard to find.

Reply:
Here's a patch file for your branch. I've also fixed a few minor other issues I just saw. In addition to your list above (which are all sattisfied), this
- does nothing if Dispose is called multiple times
- if an instance is used after Dispose, ObjectDisposedException is thrown

Reply:
Thanks, I understand now. So then, rather than raising an exception, why not then use the other pattern, for example present in some of the classes like the Explorer Hat: https://github.com/dotnet/iot/blob/3c4034781d158b26be0bc4866c34192d86bc4ef1/src/devices/ExplorerHat/Motors.cs#L87
So having a variable to avoid double disposing? Advantage: it doesn't raise an exception. I'm not much in favor of raising exceptions in the Dispose function. Those are exceptions usually no one really trap.

Reply:
The exception is not thrown in Dispose (thats what we try to avoid), but if the instance is otherwise used after Disposal, i.e when trying to read from the sensor after disposing. 
An ObjectDisposedException is rightly (almost) never handled in user code, since that is a programming error,  and they should stop the application typically in the debugger. 

Of course the same as what I did can be achieved by an extra bool variable,  but that doesn't change much conceptionally.

Reply:
OK, so I adjusted only the dispose part to make sure that is multiple dispose are called, it won't create any issue.
In case of a programming error and the class is called once disposed, an exception will be raised anyway. 
So it keeps the solution simple enough and easy to read. I hope it's a good compromise.

Reply:
I would still set it to null, but I'm ok with this as well.